# Notebook 05 — Final Load Prep

**Project:** Zomato Delivery Operations Analytics
**Sector:** Food Tech / Delivery Operations


### Objective
This notebook covers:
1. Loading the cleaned dataset
2. KPI computation
3. Final data preparation for Tableau
4. Saving the final dataset

Using cleaned dataset from 02_cleaning.ipynb

In [1]:
import pandas as pd
import numpy as np

In [2]:
# Load the cleaned dataset
df = pd.read_csv('/content/Zomato_Cleaned.csv')

print("Cleaned dataset loaded!")
print(f"Rows    : {df.shape[0]}")
print(f"Columns : {df.shape[1]}")

Cleaned dataset loaded!
Rows    : 45584
Columns : 25


In [3]:
# KPI 1 — Average Delivery Time
avg_delivery_time = df['Time_taken (min)'].mean().round(2)

print(f"Average Delivery Time : {avg_delivery_time} mins")

Average Delivery Time : 26.29 mins


In [4]:
# KPI 2 — Average Delivery Time by City
city_avg = df.groupby('City')['Time_taken (min)'].mean().round(2)

print("Average Delivery Time by City:")
print(city_avg)

Average Delivery Time by City:
City
Metropolitan    27.14
Semi-Urban      49.73
Urban           22.98
Name: Time_taken (min), dtype: float64


In [5]:
# KPI 3 — Average Delivery Time by Traffic Density
traffic_avg = df.groupby('Road_traffic_density')['Time_taken (min)'].mean().round(2)

print("Average Delivery Time by Traffic Density:")
print(traffic_avg)

Average Delivery Time by Traffic Density:
Road_traffic_density
High      27.24
Jam       31.18
Low       21.46
Medium    26.70
Name: Time_taken (min), dtype: float64


In [6]:
# KPI 4 — Average Delivery Time by Weather Conditions
weather_avg = df.groupby('Weather_conditions')['Time_taken (min)'].mean().round(2)

print("Average Delivery Time by Weather Conditions:")
print(weather_avg)

Average Delivery Time by Weather Conditions:
Weather_conditions
Cloudy        28.92
Fog           28.74
Sandstorms    25.88
Stormy        25.87
Sunny         21.86
Windy         26.12
Name: Time_taken (min), dtype: float64


In [7]:
# KPI 5 — Festival vs Non-Festival Average Delivery Time
festival_avg = df.groupby('Festival')['Time_taken (min)'].mean().round(2)

print("Average Delivery Time by Festival:")
print(festival_avg)

Average Delivery Time by Festival:
Festival
No     25.91
Yes    45.52
Name: Time_taken (min), dtype: float64


In [8]:
# KPI 6 — Average Delivery Time by Vehicle Type
vehicle_avg = df.groupby('Type_of_vehicle')['Time_taken (min)'].mean().round(2)

print("Average Delivery Time by Vehicle Type:")
print(vehicle_avg)

Average Delivery Time by Vehicle Type:
Type_of_vehicle
bicycle             26.43
electric_scooter    24.47
motorcycle          27.61
scooter             24.48
Name: Time_taken (min), dtype: float64


In [9]:
# KPI 7 — Top 10 Fastest Delivery Partners
top_partners = df.groupby('Delivery_person_ID')['Time_taken (min)'].mean().round(2).sort_values().head(10)

print("Top 10 Fastest Delivery Partners:")
print(top_partners)

Top 10 Fastest Delivery Partners:
Delivery_person_ID
KOLRES06DEL03     19.44
DEHRES20DEL03     19.60
DEHRES17DEL01     19.75
KOLRES01DEL03     19.82
KNPRES01DEL01     20.00
BHPRES15DEL03     20.14
LUDHRES18DEL02    20.20
KOLRES14DEL02     20.62
GOARES05DEL03     20.75
KOCRES16DEL03     21.00
Name: Time_taken (min), dtype: float64


In [10]:
# KPI 8 — Order Volume by Hour
order_by_hour = df.groupby('Order_Hour')['ID'].count().reset_index()
order_by_hour.columns = ['Hour', 'Order_Count']

print("Order Volume by Hour:")
print(order_by_hour)

Order Volume by Hour:
    Hour  Order_Count
0    8.0         1817
1    9.0         1770
2   10.0         1780
3   11.0         1768
4   12.0          703
5   13.0          704
6   14.0          717
7   15.0          787
8   16.0          656
9   17.0         4211
10  18.0         4077
11  19.0         4169
12  20.0         4089
13  21.0         6007
14  22.0         4172
15  23.0         4089


In [11]:
# KPI 9 — Order Volume by Day of Week
order_by_day = df.groupby('Day_of_Week')['ID'].count().reset_index()
order_by_day.columns = ['Day', 'Order_Count']

print("Order Volume by Day of Week:")
print(order_by_day)

Order Volume by Day of Week:
         Day  Order_Count
0     Friday         7028
1     Monday         6206
2   Saturday         6287
3     Sunday         6248
4   Thursday         6348
5    Tuesday         6374
6  Wednesday         7093


In [12]:
# KPI 10 — Average Delivery Person Ratings by City
ratings_by_city = df.groupby('City')['Delivery_person_Ratings'].mean().round(2)

print("Average Delivery Person Ratings by City:")
print(ratings_by_city)

Average Delivery Person Ratings by City:
City
Metropolitan    4.63
Semi-Urban      4.51
Urban           4.67
Name: Delivery_person_Ratings, dtype: float64


In [13]:
# Save final dataset for Tableau(which is basically Zomato_Cleaned but we just renamed it as Zomato_Final)
df.to_csv('/content/Zomato_Final.csv', index=False)
print(f"Rows    : {df.shape[0]}")
print(f"Columns : {df.shape[1]}")

Rows    : 45584
Columns : 25


##Final Load Prep Summary

### KPIs Computed
1. Average Delivery Time overall
2. Average Delivery Time by City
3. Average Delivery Time by Traffic Density
4. Average Delivery Time by Weather Conditions
5. Average Delivery Time by Festival
6. Average Delivery Time by Vehicle Type
7. Top 10 Fastest Delivery Partners
8. Order Volume by Hour
9. Order Volume by Day of Week
10. Average Delivery Person Ratings by City

### Output
- Final dataset saved as : Zomato_Final.csv
- Rows                   : 45,584
- Columns                : 25

### Next Step
Build Tableau dashboard using Zomato_Final.csv